example/oven_fsm.dart

```diff
@@ -0,0 +1,228 @@
+            counterReset < 0,
+            en < 1,
+          ]),
+
+      // identifier: paused state, represent by `OvenState.paused`.
+      State<OvenState>(OvenState.paused,
+          // events:
+          // When the button `resume` is pressed during paused state,
+          // OvenState will changed to `OvenState.cooking` state.
+          events: {
+            Logic(name: 'button_resume')..gets(button.eq(Button.resume())):
+                OvenState.cooking
+          },
+          // actions:
+          // During the paused state, `led` is change to red; timer's
+          // `counterReset` is set to 0 (Do not reset);
+          // timer's `en` is set to 0 (Disable value update).
+          actions: [
+            led < LEDLight.red().value,
+            counterReset < 0,
+            en < 0,
+          ]),
+
+      // identifier: completed state, represent by `OvenState.completed`.
+      State<OvenState>(OvenState.completed,
+          // events:
+          // When the button `start` is pressed during completed state,
+          // OvenState will changed to `OvenState.cooking` state.
+          events: {
+            Logic(name: 'button_start')..gets(button.eq(Button.start())):
+                OvenState.cooking
+          },
+          // actions:
+          // During the start state, `led` is change to green; timer's
+          // `counterReset` is set to 1 (Reset value);
+          // timer's `en` is set to 0 (Disable value update).
+          actions: [
+            led < LEDLight.green().value,
+            counterReset < 1,
+            en < 0,
+          ])
+    ];
+
+    // Assign the oven StateMachine object to public variable declared.
+    oven = StateMachine<OvenState>(clk, reset, OvenState.standby, states);
+  }
+
+  // An ovenStateMachine that represent in getter.
+  StateMachine<OvenState> get ovenStateMachine => oven;
+}
+
+Future<void> main() async {
+  // Signals `button` and `reset` that mimic user's behaviour of button pressed
+  // and reset.
+  //
+  // Width of button is 2 because button is represent by 2-bits signal.
+  final button = Logic(name: 'button', width: 2);
+  final reset = Logic(name: 'reset');
+
+  // Build an Oven Module and passed the `button` and `reset`.
+  final oven = OvenModule(button, reset);
+
+  // Generate a Mermaid FSM diagram and save as the name `oven_fsm.md`.
+  // Note that the extension of the files is recommend as .md or .mmd.
+  //
+  // Check on https://mermaid.js.org/intro/ to view the diagram generated.
+  // If you are using vscode, you can download the mermaid extension.
+  oven.ovenStateMachine.generateDiagram(outputPath: 'oven_fsm.md');
+
+  // Before we can simulate or generate code with the counter, we need
+  // to build it.
+  await oven.build();
+
+  // Now let's try simulating!
+
+  // Let's start off with asserting reset to Oven.
+  reset.inject(1);
+
+  // Attach a waveform dumper so we can see what happens.
+  WaveDumper(oven, outputPath: 'oven.vcd');
+
```

Comment:
It could be fun to register something listening to the LED (`led.changed.listen...`) so that without looking at the waves you can still see the oven going through different states.